src/app/auth/signup/page.tsx

```diff
@@ -0,0 +1,183 @@
+'use client';
+import React, { use, useState } from 'react';
+import axios from 'axios';
+import Image from 'next/image';
+import InputBox from '@/components/InputBox';
+import Button from '@/components/Button';
+import PopUpModels from '@/components/PopUpModels';
+import { signUpVAlidation } from '@/validations/validationSchema';
+import { unknown } from 'zod';
+
+const Page = () => {
+  const initial = {
+    firstName: '',
+    lasttName: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  };
+  const [showModal, setShowmodal] = useState(false);
```

Comment:
@tchamianest separate your logic from UI components, to achieve this you can use custom hooks as you learned during capstone project or use server action (documentation available on Nextjs doc). Also see how you can make your pages server side.